pkg/controller/certificates/certificate_request.go

```diff
@@ -531,51 +531,86 @@ func (c *certificateRequestManager) processCertificate(ctx context.Context, crt
 		return nil
 	}
 
-	// Check if the CertificateRequest is Ready, if it is not then we return
-	// and wait for informer updates to re-trigger processing.
-	if !apiutil.CertificateRequestHasCondition(existingReq, cmapi.CertificateRequestCondition{
-		Type:   cmapi.CertificateRequestConditionReady,
-		Status: cmapi.ConditionTrue,
-	}) {
-		log.Info("certificate request is not in a Ready state, waiting until CertificateRequest is issued")
-		// TODO: we need to handle failure states too once we have defined how we represent them
-		return nil
-	}
+	// Determine the status reason of the CertificateRequest and process accordingly
+	switch apiutil.CertificateRequestStatusReason(existingReq) {
+	// If the CertificateRequest exists but has failed then we check the failure
+	// time. If the failure time doesn't exist or is over an hour in the past
+	// then delete the request so it can be re-created on the next sync. If the
+	// failure time is less than an hour in the past then schedule this owning
+	// Certificate for a re-sync in an hour.
+	case cmapi.CertificateRequestReasonFailed:
+		if existingReq.Status.FailureTime == nil {
+			log.Info("the failed existing certificate request has no failure time so will be retried now")
+
+		} else if c.clock.Since(existingReq.Status.FailureTime.Time) > time.Hour {
+			log.Info("the failed existing certificate request failed over an hour ago so will be retried now")
+
+		} else {
+			log.Info("the failed existing certificate request failed less than an hour ago, will be scheduled for reprocessing in an hour")
 
-	log.Info("CertificateRequest is in a Ready state, issuing certificate...")
+			key, err := keyFunc(crt)
+			if err != nil {
+				log.Error(err, "error getting key for certificate resource")
+				return nil
+			}
 
-	// Decode the certificate bytes so we can ensure the certificate is valid
-	log.Info("decoding certificate data")
-	x509Cert, err := pki.DecodeX509CertificateBytes(existingReq.Status.Certificate)
-	if err != nil {
-		return err
-	}
+			// We don't fire an event here as this could be called multiple times in quick succession
```

Comment:
It may be worth still logging a message however? For those trying to understand wtf cert-manager is doing when things go wrong :fire:

Reply:
We do a couple lines up